src/Functions/array/arraySplit.cpp

```diff
@@ -115,8 +115,35 @@ using FunctionArrayReverseSplit = FunctionArrayMapped<ArraySplitImpl<true>, Name
 
 REGISTER_FUNCTION(ArraySplit)
 {
-    factory.registerFunction<FunctionArraySplit>();
-    factory.registerFunction<FunctionArrayReverseSplit>();
+    FunctionDocumentation::Description description = "Split a source array into multiple arrays. When `func(x [, y1, ..., yN])` returns something other than zero, the array will be split on the left hand side of the element. The array will not be split before the first element.";
+    FunctionDocumentation::Syntax syntax = "arraySplit(func(x[, y1, ..., yN]), source[, cond1, ... , condN])";
+    FunctionDocumentation::Arguments arguments = {
+        {"func(x[, y1, ..., yN])", "A lambda function which operates on elements of the source array (`x`) and condition arrays (`y`).[Lambda function](/sql-reference/functions/overview#arrow-operator-and-lambda)."},
+        {"source", "The source array to split [`Array(T)`](/sql-reference/data-types/array)."},
+        {"[, cond1, ... , condN]", "Optional. N condition arrays providing additional arguments to the lambda function. [`Array(T)`](/sql-reference/data-types/array)."}
+    };
+    FunctionDocumentation::ReturnedValue returned_value = "Returns an array of arrays. [`Array(Array(T))`](/sql-reference/data-types/array).";
+    FunctionDocumentation::Examples examples = {{"Usage example", "SELECT arraySplit((x, y) -> y, [1, 2, 3, 4, 5], [1, 0, 0, 1, 0]) AS res", "[[1, 2, 3], [4, 5]]"}};
+    FunctionDocumentation::IntroducedIn introduced_in = {20, 1};
+    FunctionDocumentation::Category category = FunctionDocumentation::Category::Array;
+    FunctionDocumentation documentation = {description, syntax, arguments, returned_value, examples, introduced_in, category};
+
+    factory.registerFunction<FunctionArraySplit>(documentation);
+
+    FunctionDocumentation::Description description_split = "Split a source array into multiple arrays. When `func(x[, y1, ..., yN])` returns something other than zero, the array will be split on the right hand side of the element. The array will not be split after the last element.";
+    FunctionDocumentation::Syntax syntax_split = "arrayReverseSplit(func(x[, y1, ..., yN]), source[, cond1, ... , condN])";
+    FunctionDocumentation::Arguments arguments_split = {
+        {"func(x[, y1, ..., yN])", "Optional. A lambda function which operates on elements of the source array (`x`) and condition arrays (`y`). [Lambda function](/sql-reference/functions/overview#arrow-operator-and-lambda)."},
```

Comment:
Argument `func` is not optional.

Reply:
Could we rename `source` to `source_arr` and use it instead of `x`?

Similarly, we should rename `cond1, ..., condN` to `cond_arr1, ..., cond_arrN` and use this terminology instead of `y1, ..., yN`.